src/main/resources/sql/insertDiscreteGroundWaterData.sql

```diff
@@ -53,8 +53,21 @@ with upd as (
           and field_visit_header_info.partition_number = ?
         returning location_identifier
 ) select
-         location_identifier,
+        (
+            select
+                location_identifier
+            from field_visit_header_info
+            where json_data_id = ?
+              and field_visit_header_info.partition_number = ?
+                fetch first 1 rows only
+        ) location_identifier,
+        (
+            select
+                coalesce(nullif((regexp_match(location_identifier, '(\d*)-*(.*)'))[2], ''), 'USGS') || '-' || (regexp_match(location_identifier, '(\d*)-*(.*)'))[1] monitoring_location_identifier
+            from field_visit_header_info
+            where json_data_id = ?
+              and field_visit_header_info.partition_number = ?
+                fetch first 1 rows only
+        ) monitoring_location_identifier,
```

Comment:
If there is a better way to do this query, I am all ears.

Reply:
Do not know a particularity better way at the moment, as long as it produces the same value monitoring_location_identifier that is setting in the monitoring_location_identifier table. it is good for now.
If it is about speeding, the query I might have a suggestion or two...